level_5/interpreter.hpp

```diff
@@ -0,0 +1,81 @@
+#pragma once
+#include <iostream>
+#include <stdexcept>
+#include "lexer.hpp"
+
+class Interpreter{
+	private:
+		Lexer lexer_;
+		Token current_token_;
+		void error_(){
+			throw std::logic_error("Invalid syntex");
+		}
+	public:
+		/// Interpreter constructor for the calculator
+		///
+		/// \param lexer Lexer for getting the tokens, one at a time
+		Interpreter(const Lexer& lexer) : lexer_(lexer), current_token_(lexer_.getNextToken())
+		{}
+		/// Eat the current token
+		///
+		/// Compare the *current_token_* with the one that is passed, if they both match,
+		/// The symtex is currect, If not, it is a syntex error and we throw an exception
+		/// \param token_type Type to check against
+		void eat(Type token_type){
+			if(this->current_token_.getToken() == token_type)
```

Comment:
Why all this `this->` if private member variables already encode suffix `_` underscore?